src/main/java/bio/terra/service/dataset/flight/delete/DatasetDeleteFlight.java

```diff
@@ -59,10 +61,17 @@ public DatasetDeleteFlight(FlightMap inputParameters, Object applicationContext)
 
     addStep(new LockDatasetStep(datasetDao, datasetId, false, true), lockDatasetRetry);
     addStep(new DeleteDatasetValidateStep(snapshotDao, dependencyDao, datasetService, datasetId));
-    addStep(
-        new DeleteDatasetPrimaryDataStep(
-            bigQueryPdao, gcsPdao, fileDao, datasetService, datasetId, configService),
-        primaryDataDeleteRetry);
+    if (platform.isGcp()) {
+      addStep(
+          new DeleteDatasetPrimaryDataStep(
```

Comment:
For consistency's sake, should this be renamed `DeleteDatasetGcpPrimaryDataStep`? If instead `DeleteDatasetPrimaryDataStep` was modified to handle both Azure and GCP, then its name wouldn't need to be changed.

Reply:
I didn't want to rename because I think that may have unintended consequences when we try to deserialize an old flight.

Re: combining in the steps, I wanted to be consistent with branching at the flight level

Reply:
There should be a way for us to rename flights and preserve backward compatibility. Adding both names and having the old one be `Deprecated` would work. It's not a big deal in this case, but we shouldn't need to avoid changing our code for this reason only.

Regarding consistency of conditionalizing our code, I think it's still in the early stages so it's fair to think about doing things differently. In the previous cases I've seen, we don't have common code between the GCP and azure steps, so there wasn't a benefit in doing the test in the step vs in the flight. As long as it's clear that a step is supporting both GCP and Azure, I don't see a problem with having a step do the check instead of a flight, if it makes sense to do so.

Reply:
Yeah, though looking ahead, there's value to having the flight (and its steps) clearly represent the path that it's running (e.g. Azure vs. GCP) since a lot of steps get outright ignored in some cloud platforms vs. others.